src/__tests__/ParserComponent.test.tsx

```diff
@@ -436,6 +439,91 @@ describe('<ParserComponent /> Unit Tests', () => {
+      instance.setState({formatString: 'DD-MM-YYYY'});
+      expect(instance.lookForDateKey(testArray)).toEqual('Date');
+    });
+    it('should return column name with the valid date in the first row' +
+        'in date format DD-MMMM-YYYY', () => {
+      const testArray: {Name: string, Date: string}[] = [
+        {'Name': 'name', 'Date': '21 November 2019'}];
+      instance.setState({formatString: 'DD-MMMM-YYYY'});
+      expect(instance.lookForDateKey(testArray)).toEqual('Date');
+    });
+    it('should return column name with the valid date in the first row' +
+        'in date format MMMM-DD-YYYY', () => {
+      const testArray: {Name: string, Date: string}[] = [
+        {'Name': 'name', 'Date': 'November 21, 2019'}];
+      instance.setState({formatString: 'MMMM-DD-YYYY'});
+      expect(instance.lookForDateKey(testArray)).toEqual('Date');
+    });
+    it('should return column name with the valid date in the first row' +
+        'in date format MMM-DD-YYYY', () => {
+      const testArray: {Name: string, Date: string}[] = [
+        {'Name': 'name', 'Date': 'Feb 21, 2019'}];
+      instance.setState({formatString: 'MMM-DD-YYYY'});
+      expect(instance.lookForDateKey(testArray)).toEqual('Date');
+    });
+    it('should return column name with the valid date in the first row' +
+        'in date format DD-MM-YY', () => {
+      const testArray: {Name: string, Date: string}[] = [
+        {'Name': 'name', 'Date': '12/2/18'}];
+      instance.setState({formatString: 'DD-MM-YY'});
+      expect(instance.lookForDateKey(testArray)).toEqual('Date');
+    });
+    it('should return column name with the valid date in the first row' +
+        'in date format MM-DD-YY', () => {
+      const testArray: {Name: string, Date: string}[] = [
+        {'Name': 'name', 'Date': '12/2/19'}];
+      instance.setState({formatString: 'MM-DD-YY'});
+      expect(instance.lookForDateKey(testArray)).toEqual('Date');
+    });
+    it('should return undefined when given data with no date', () => {
+      const testArray: {Name: string, AnotherName: string}[] = [
+        {'Name': 'name', 'AnotherName': 'Name'}];
+      expect(instance.lookForDateKey(testArray)).toEqual(undefined);
+    });
+  });
+
+  describe('dataIsValid()', () => {
+    const wrapper = shallow(<ParserComponent prompt={'Select ' +
+    'a CSV file: '} fileType={FileType.csv}
+    onChange={function() {
+    }}/>);
+    const instance = wrapper.instance() as ParserComponent;
+    it('should return true when given array that has data ', () => {
+      const testArray: { Date: string }[] = [
+        {'Date': '4/5/2010'},
+        {'Date': '2/31/1992'},
+        {'Date': '12/21/1992'}];
+      expect(instance.dataIsValid(testArray)).toBeTruthy();
+    });
+    it('should return false when given empty array', () => {
+      const testArray: {}[] = [];
+      try {
+        expect(instance.dataIsValid(testArray)).toThrow('File has no data');
+      } catch (e) {
+        console.log(e);
```

Comment:
I don't understand this catch. You check if something was thrown in the previous call so this doesn't seem necessary.